src/utils/util.ts

```diff
@@ -40,3 +41,34 @@ export const isDataEqual = (
 ): boolean => {
   return is(oldData, newData);
 };
+
+export const paginate = (
+  list: List<RecordOf<any>>,
+  pageSize: number,
+  pageNumber: number,
+): Promise<RecordOf<any>> => {
+  return new Promise((resolve, reject) => {
+    try {
+      const data = list
+        .filter((i) => i.type !== ITEM_TYPES.FOLDER)
+        .filter((i) => !i.settings?.isPinned)
```

Comment:
These are conditions only valid for player in the right-hand side of the view. Perhaps you could pass some `conditions` to the hook to be applied here. Also, it might be better to check both conditions at once? (avoid a filter)